reporting-anything/src/download_helpers.js

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import puppeteer from 'puppeteer-core';
+import fs from 'fs';
+import { JSDOM } from 'jsdom';
+import { CHROMIUM_PATH, FORMAT, REPORT_TYPE, SELECTOR } from './constants.js';
+
+
+export async function downloadVisualReport(url, format, width, height, filename, authType, username, password) {
+  const window = new JSDOM('').window;
+
+  try {
+    const browser = await puppeteer.launch({
+      headless: true,
+      /**
+       * TODO: temp fix to disable sandbox when launching chromium on Linux instance
+       * https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md#setting-up-chrome-linux-sandbox
+       */
+      args: [
+        '--no-sandbox',
+        '--disable-setuid-sandbox',
+        '--disable-gpu',
+        '--no-zygote',
+        '--single-process',
+        '--font-render-hinting=none',
+        '--enable-features=NetworkService',
+        '--ignore-certificate-errors',
+      ],
+      executablePath: CHROMIUM_PATH,
+      ignoreHTTPSErrors: true,
+      env: {
+        TZ: 'UTC', // leave as UTC for now
+      },
+    });
+
+    const page = await browser.newPage();
+    const page2 = await browser.newPage();
+    page.setDefaultNavigationTimeout(0);
+    page.setDefaultTimeout(100000);
+    page2.setDefaultNavigationTimeout(0);
+    page2.setDefaultTimeout(100000);
+
+    // auth 
+    if (username !== undefined && password !== undefined) {
+      if(authType === 'basic'){
+        console.log(authType);
+        await page.goto(url, { waitUntil: 'networkidle0' });
+        console.log('authenticating');
+        await page.waitFor(10000);
+        await page.type('input[data-test-subj="user-name"]', username);
```

Comment:
Can we extract some of these to functions? There are a lot of statements here and it can get dense to read.

e.g something like `enterUsername(username)`